code/game/objects/items/toys.dm

```diff
@@ -50,22 +51,9 @@
 	return
 
 /obj/item/toy/waterballoon/attackby(obj/O as obj, mob/user as mob)
-	if(istype(O, /obj/item/weapon/reagent_containers/glass))
-		if(O.reagents)
-			if(O.reagents.total_volume < 1)
-				to_chat(user, "The [O] is empty.")
-			else if(O.reagents.total_volume >= 1)
-				if(O.reagents.has_reagent(PACID, 1))
-					to_chat(user, "The acid chews through the balloon!")
-					O.reagents.reaction(user)
-					qdel(src)
-					return
-				else
-					src.desc = "A translucent balloon with some form of liquid sloshing around in it."
-					to_chat(user, "<span class = 'info'>You fill the balloon with the contents of \the [O].</span>")
-					O.reagents.trans_to(src, 10)
-	src.update_icon()
-	return
+	if(..())
```

Comment:
What does any of this do?

Reply:
its so that if it does the attackby, it will update its icon. the removed stuff was redundant due to opencontainer flag.

Reply:
As far as I can tell `toy/attackby` is not defined, `item/attackby` merely supercalls, and `obj/attackby` doesn't return anything. So this will never do anything, right?